lib/cache/add-remote-git.js

```diff
@@ -26,146 +26,149 @@ var VALID_VARIABLES = [
+  var reconstitutedURL
+  if (!/^git[+:]/.test(gitURL)) {
+    reconstitutedURL = 'git+' + gitURL
+  } else {
+    reconstitutedURL = gitURL
+  }
+  var params = {
+    silent: silent,
+    gitURL: strippedURL,
+    originalURL: reconstitutedURL,
+    treeish: treeish,
+    checkoutPath: checkoutPath
+  }
+  log.verbose('addRemoteGit', 'params', params)
 
-  var p = path.join(remotes, v)
-  cb = inflight(p, cb)
-  if (!cb) return log.verbose("addRemoteGit", p, "already in flight; waiting")
-  log.verbose("addRemoteGit", p, "not in flight; cloning")
+  cb = inflight(checkoutPath, cb)
+  if (!cb) {
+    return log.verbose('addRemoteGit', checkoutPath, 'already in flight; waiting')
+  }
+  log.verbose('addRemoteGit', checkoutPath, 'not in flight; caching')
 
   getGitDir(function (er) {
     if (er) return cb(er)
-    checkGitDir(p, normalized.url, normalized.branch, u, silent, function (er, data) {
+    checkGitDir(params, function (er, data) {
       if (er) return cb(er, data)
 
-      addModeRecursive(p, npm.modes.file, function (er) {
+      addModeRecursive(checkoutPath, npm.modes.file, function (er) {
         return cb(er, data)
       })
     })
   })
 }
 
-function getGitDir (cb) {
-  getCacheStat(function (er, st) {
-    if (er) return cb(er)
-
-    // We don't need global templates when cloning. Use an empty directory for
-    // the templates, creating it (and setting its permissions) if necessary.
-    mkdir(templates, function (er) {
-      if (er) return cb(er)
-
-      // Ensure that both the template and remotes directories have the correct
-      // permissions.
-      fs.chown(templates, st.uid, st.gid, function (er) {
-        if (er) return cb(er)
-
-        fs.chown(remotes, st.uid, st.gid, function (er) {
-          cb(er, st)
-        })
-      })
-    })
-  })
-}
-
-function checkGitDir (p, u, co, origUrl, silent, cb) {
-  fs.stat(p, function (er, s) {
-    if (er) return cloneGitRemote(p, u, co, origUrl, silent, cb)
-    if (!s.isDirectory()) return rm(p, function (er) {
+function checkGitDir (params, cb) {
+  fs.stat(params.checkoutPath, function (er, s) {
+    if (er) return cloneGitRemote(params, cb)
+    if (!s.isDirectory()) return rm(params.checkoutPath, function (er) {
       if (er) return cb(er)
-      cloneGitRemote(p, u, co, origUrl, silent, cb)
+      cloneGitRemote(params, cb)
     })
 
     git.whichAndExec(
-      [ "config", "--get", "remote.origin.url" ],
-      { cwd : p, env : gitEnv() },
+      ['config', '--get', 'remote.origin.url'],
+      { cwd: params.checkoutPath, env: gitEnv() },
       function (er, stdout, stderr) {
-        var stdoutTrimmed = (stdout + "\n" + stderr).trim()
-        if (er || u !== stdout.trim()) {
-          log.warn( "`git config --get remote.origin.url` returned "
-                  + "wrong result ("+u+")", stdoutTrimmed )
-          return rm(p, function (er){
+        var originURL = (stdout + '\n' + stderr).trim()
+        if (er || params.gitURL !== stdout.trim()) {
```

Comment:
That was how it was written originally, which probably means that it's a byproduct of how @isaacs originally hacked it together, and yes, I think its use is purely informational. This particular check is just verifying that the repo we've been handled is for the source we're expecting. I'll try cleaning this up and see if anything breaks.